x-pack/plugins/security/server/authorization/disable_ui_capabilities.ts

```diff
@@ -105,7 +105,11 @@ export function disableUICapabilitiesFactory(
       }
 
       const action = authz.actions.ui.get(featureId, ...uiCapabilityParts);
-      return checkPrivilegesResponse.privileges[action] === true;
+      return (
+        checkPrivilegesResponse.privileges.find(
+          x => x.privilege === action && x.authorized === true
+        ) !== undefined
+      );
```

Comment:
nit: you can simplify this with [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some):

```suggestion
      return checkPrivilegesResponse.privileges.some(
        x => x.privilege === action && x.authorized === true
      );
```